lib/chef/provider/service/windows.rb

```diff
@@ -281,6 +266,23 @@ def action_configure_startup
 
   private
 
+  def configure_service_run_as_properties
+    return unless new_resource.property_is_set?(:run_as_user)
```

Comment:
This is the actual fix to #8080

Reply:
is `return unless new_resource.run_as_user == "LocalSystem"` the same?  its just that `property_is_set?` checking is a bit odd, and that creates the need to go look up what the default was.  also, if someone sets the property to "LocalSystem" should it behave the same way here (in which case it may actually be a bug you've written here?).

if the user specifying "LocalSystem" is different than the default of "LocalSystem" that seems, at any rate, odd and worthy of some commented documentation if that's the way things are...

Reply:
I'm asserting that the behavior _should_ be different whether it's the default value versus being explicitly set to that same default value. The logic behind this is that a chef user may desire to use the `windows_service` just for notifications, `:start`/`:stop`/`:restart`, etc.

I believe the current behavior is unexpected. I don't believe that someone with the following resource would actually want the service to be re-configured. This is especially true for built-in/native windows services that are drivers which don't even run as a specific user (read reference below). With that said, this is definitely in need of some clear documentation because it's going to be confusing regardless.

```ruby
windows_service 'Dhcp' do
  action :start
end
```

> If the service type is SERVICE_KERNEL_DRIVER or SERVICE_FILE_SYSTEM_DRIVER, the name is the driver object name that the system uses to load the device driver. Specify NULL if the driver is to use a default object name created by the I/O system.

Reference: https://docs.microsoft.com/en-us/windows/desktop/api/winsvc/nf-winsvc-changeserviceconfiga